src/query/frontend/semantic/uses_database.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "query/frontend/ast/ast.hpp"
+#include "query/frontend/ast/ast_visitor.hpp"
+#include "query/procedure/module.hpp"
+#include "utils/memory.hpp"
+
+namespace memgraph::query {
+
+class BLA : public QueryVisitor<void>, public HierarchicalTreeVisitor {
+ public:
+  using HierarchicalTreeVisitor::PostVisit;
+  using HierarchicalTreeVisitor::PreVisit;
+  using HierarchicalTreeVisitor::Visit;
+  using QueryVisitor<void>::Visit;
+
+  bool state() { return state_; }
+
+  bool PreVisit(Create & /*unused*/) override {
```

Comment:
Can you expkain this usage_kind a bit more?

Reply:
I think this needs cleaning up.
Not sure if we ended up depending on it or not.
Its a visitor which would say if what kind of query we have and hence what the interpreter should do.
ie. does it need DB access or not + does it need DB transactional access or not + will that DB transactional access perform writes (aka expected to do work on commit)

I don't think we got around to completing this/depending on it.